### PR TITLE
ENH: test-datalad: Enable SSH testing

### LINKED ISSUE #19: Add setup (docker) and enable ssh testing
Should be done within https://github.com/datalad/datalad-extensions/blob/master/.github/workflows/build-git-annex-debianstandalone.yaml only for now, until #18 is merged.

@kyleam, help?!

### PR DESCRIPTION
**Pull request by kyleam**
Make use of the Docker SSH target that DataLad supports as of
datalad/datalad#4762.

Closes #19.

---

Replaces #26.  As I say there, I suspect this won't work as is, and setting up the docker container will require more tweaking.

### CODE REVIEW

**Comment by kyleam** [comment]
In terms of test failures, that looked pretty good.  One SSH test (`test_create_sibling.test_target_ssh_inherit('manual', True, True)`) failed in both the master and maint jobs due to chgrp permission issues: https://github.com/datalad/datalad-extensions/runs/920594967

I haven't looked at the failure closely, but that's probably something to be adjusted in the `setup` script of the docker repo.

**Comment by kyleam** [comment]
All the tests are now in a passing state.  

The run time for test-datalad is ~1.5h.

File: .github/workflows/build-git-annex-debianstandalone.yaml [hunk 1]
```diff
@@ -124,6 +124,17 @@ jobs:
         run: |
           sudo dpkg -i git-annex*.deb
 
+      - name: Set up SSH target
+        shell: bash
+        # TODO: Drop the release condition once 0.13.1 is released.
```
Comment (yarikoptic) [inline .github/workflows/build-git-annex-debianstandalone.yaml:129]: did you mean `0.13.2`?

File: .github/workflows/build-git-annex-debianstandalone.yaml [hunk 2]
```diff
@@ -124,6 +124,17 @@ jobs:
...
+        run: |
+          if [ "${{ matrix.version }}" != "release" ]; then
+            curl -fSsL \
+              https://raw.githubusercontent.com/datalad/datalad/master/tools/ci/prep-travis-forssh.sh \
+              | bash;
+            echo "::set-env name=DATALAD_TESTS_SSH::1";
```
Comment (yarikoptic) [inline .github/workflows/build-git-annex-debianstandalone.yaml:135]: is that some github actions specific way to export env vars from one step into another and regular `export DATALAD_TESTS_SSH=1` wouldn't work?

File: .github/workflows/build-git-annex-debianstandalone.yaml (hunk 1 above)
Comment (kyleam) [reply .github/workflows/build-git-annex-debianstandalone.yaml:129]: Yes, thanks for catching that.

File: .github/workflows/build-git-annex-debianstandalone.yaml (hunk 2 above)
Comment (kyleam) [reply .github/workflows/build-git-annex-debianstandalone.yaml:135]: I tried `export DATALAD_TESTS_SSH=1` at first and the SSH tests were skipped, so it seems like those values don't persist in subsequent steps.

The snippet above was the first thing I [stumbled on][0] that worked, though there might be better ways to do it.

[0]: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable

Once the released version of DataLad has the SSH test changes, I think we can just set it via `env`, though.

**Comment by kyleam** [comment]
> I am also confused by commit msg a bit: "Note that DataLad's master version of prep-travis-forssh.sh is used unconditionally." -- did you mean to say "conditionally"?

No, I meant "unconditionally", but it's confusing word choice since there is currently a `[ "${{ matrix.version }}" != "release" ]` condition.

```
Note that DataLad's master version of prep-travis-forssh.sh is used
unconditionally.  That's simple and should work fine for maint and the
latest release (once a release contains the changes from PR 4762).
But if this approach ends up causing problems, we can of course switch
to a more specific download or to cloning the repo and checking out
the appropriate revision.
```

That's referring to using `master/tools/ci/prep-travis-forssh.sh` not only for master but also for maint and (eventually) the release, as opposed to using `master/tools/...` for master, `maint/tools/...` for maint, and (eventually) `<rel>/tools/...` for the last release.

> I guess the next step would be to add the same to extensions environments? I would not bother annotating for which specifically need it it or not, just add to the template, regenerate and go

Okay.

**Comment by kyleam** [comment]
> I guess the next step would be to add the same to extensions environments? I would not bother annotating for which specifically need it it or not, just add to the template, regenerate and go

I pushed that.  However, as far as I can tell by grepping for `skip_ssh`, -metalad is the only extension that has an SSH test.  And it looks like that needs to `s/localhost/datalad-test/` to work with the Docker target.  I'll propose that on -metalad's side.

So, I'd say it makes sense to drop the extension changes for now.  Even when a new -metalad is released, it feels a bit wasteful to download the docker image for all extensions when only one extension needs it and for only a single test.

**Comment by kyleam** [comment]
> So, I'd say it makes sense to drop the extension changes for now.

Doing now...

**Comment by yarikoptic** [comment]
woohoo -- it is green (although quite length now), and the failing one is solely due to https://github.com/datalad/datalad/issues/4812 which is unrelated to this PR. Thank you @kyleam , let's proceed!

**Pull request merged**